### PR TITLE
Add event participants count and committees members count by month with filtering 

### PR DESCRIPTION
**Pull request by illusionOfParadise**
https://app.clubhouse.io/enmarche/story/4121/2e-partie-section-comit%C3%A9s-scroll-down-double-graphique-choisir-un-comit%C3%A9-2

### CODE REVIEW

File: features/api/committees.feature [hunk 1]
```diff
@@ -74,3 +75,121 @@ Feature:
...
+
+  Scenario: As a referent I can get the committee members count in referent managed zone
+    When I am logged as "referent-75-77@en-marche-dev.fr"
+    And I am on "/api/committees/members/count-by-month"
+    Then the response status code should be 200
+    And the response should be in JSON
+    And the JSON should be equal to:
+    """
+    {
+      "2018-04": {"committee_memebers":3},
```
Comment (B-Galati) [inline features/api/committees.feature]: Typo ~~`committee_memebers`~~ `committee_members`

File: src/Repository/AdherentRepository.php [hunk 2]
```diff
@@ -590,6 +591,31 @@ public function countMembersByGenderForReferent(Adherent $referent): array
...
+            ->innerJoin('committee.referentTags', 'tag')
+            ->where('tag IN (:tags)')
+            ->andWhere('committee.status = :status')
+            ->andWhere('membership.joinedAt >= :from')
+            ->andWhere('membership.joinedAt <= :until')
+            ->setParameter('tags', $referent->getManagedArea()->getTags())
+            ->setParameter('status', Committee::APPROVED)
+            ->setParameter('until', (new Chronos('now'))->setTime(23, 59, 59, 999))
+            ->setParameter('from', (new Chronos("first day of -$months months"))->setTime(0, 0, 0, 000))
+            ->groupBy('yearmonth')
```
Comment (B-Galati) [inline src/Repository/AdherentRepository.php]: Let's add a little bit of cache WDYT? Like 1 hour or 30 minutes?

File: src/Repository/EventRegistrationRepository.php [hunk 3]
```diff
@@ -163,6 +168,30 @@ public function findByRegisteredEmailAndEvent(string $email, BaseEvent $event):
         ;
     }
 
+    public function countEventPariticipantsInReferentManagedArea(Adherent $referent, StatisticsParametersFilter $filter = null, int $months = 5): array
```
Comment (B-Galati) [inline src/Repository/EventRegistrationRepository.php]: typo ~~`Pariticipants`~~ `Participants`

File: src/Repository/ReferentTrait.php [hunk 4]
```diff
@@ -12,4 +15,46 @@ private function checkReferent(Adherent $referent): void
             throw new \InvalidArgumentException('Adherent must be a referent.');
         }
     }
+
+    protected function aggregateCountByMonth(array $itemsCount, string $type, int $months = 6): array
```
Comment (B-Galati) [inline src/Repository/ReferentTrait.php]: These 2 new methods are not really related to Referent stuff.
I don't have a lot of idea : We keep these like this, we move them to static class, we move them to a properly named trait.

File: src/Repository/EventRegistrationRepository.php [hunk 5]
```diff
@@ -163,6 +168,30 @@ public function findByRegisteredEmailAndEvent(string $email, BaseEvent $event):
...
+    {
+        $this->checkReferent($referent);
+
+        $query = $this->createQueryBuilder('event_registrations')
+            ->select('DISTINCT event_registrations.emailAddress, COUNT(DISTINCT event_registrations) AS count, YEAR_MONTH(event.beginAt) as yearmonth')
```
Comment (B-Galati) [inline src/Repository/EventRegistrationRepository.php]: We need to add some indexes to improve performances of this query no ?

File: src/Repository/AdherentRepository.php (hunk 2 above)
Comment (illusionOfParadise) [reply src/Repository/AdherentRepository.php]: Yes, I add 1 hour.

File: src/Repository/ReferentTrait.php (hunk 4 above)
Comment (illusionOfParadise) [reply src/Repository/ReferentTrait.php]: I'm agree. This trait contains different functions used in many repository classes. What about renaming this trait to `RepositoryTrait` or transforming to `RepositoryUtils` static class?

**Pull request merged**